### PR TITLE
changed ssh connection type

### PR DESCRIPTION
**Pull request by Hidanio**

### CODE REVIEW

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt [hunk 1]
```diff
@@ -10,44 +10,50 @@ import com.vk.admstorm.notifications.AdmNotification
 object ChangeSshBackendStartup {
     private val LOG = logger<ChangeSshBackendStartup>()
 
-    private const val SSH_NOTIFICATOR_OPTION = "sshLegacyChanges"
+    private const val LEGACY_SSH_NOTIFICATOR_OPTION = "sshLegacyChanges"
     private const val SSH_CONFIG_BACKEND = "ssh.config.backend"
+    private const val SSH_NOTIFICATOR_OPTION = "sshOpenSSHChanges"
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:15]: Let's move it higher to be next to `LEGACY_SSH_NOTIFICATOR_OPTION`

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt [hunk 2]
```diff
@@ -10,44 +10,50 @@ import com.vk.admstorm.notifications.AdmNotification
...
 
     private fun changeSshConfiguration(configValue: SshConnectionConfigService.Kind) {
         AdvancedSettings.setEnum(SSH_CONFIG_BACKEND, configValue)
     }
 
     fun changeConfigurationProcess(project: Project) {
+        //TODO: Remove it after 2 month
+        if (PropertiesComponent.getInstance(project).getValue(LEGACY_SSH_NOTIFICATOR_OPTION) != null) {
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:23]: Why not `isValueSet`?

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt [hunk 3]
```diff
@@ -10,44 +10,50 @@ import com.vk.admstorm.notifications.AdmNotification
...
+            PropertiesComponent.getInstance(project).unsetValue(LEGACY_SSH_NOTIFICATOR_OPTION)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:24]: We do `PropertiesComponent.getInstance(project)` 5 times. Let's do it 1 time.

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt [hunk 4]
```diff
@@ -10,44 +10,50 @@ import com.vk.admstorm.notifications.AdmNotification
...
+            LOG.info("OPENSSH option was enabled")
             return
         }
 
         val sshSettingValue = AdvancedSettings.getEnum(SSH_CONFIG_BACKEND, SshConnectionConfigService.Kind::class.java)
-        if (sshSettingValue == SshConnectionConfigService.Kind.LEGACY) {
-            PropertiesComponent.getInstance(project).setValue(SSH_NOTIFICATOR_OPTION, true)
-            LOG.info("LEGACY SSH option was enabled before")
+        if (sshSettingValue == SshConnectionConfigService.Kind.OPENSSH) {
+            PropertiesComponent.getInstance(project).setValue(LEGACY_SSH_NOTIFICATOR_OPTION, true)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:35]: Why `LEGACY_SSH_NOTIFICATOR_OPTION` and not `SSH_NOTIFICATOR_OPTION`?

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt (hunk 2 above)
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:23]: I would move it after `dntShow`

Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:23]: It seems to me that this violates semantics. We want to check whether the property itself exists, not the value. In this case, if such property does not exist, then null is returned

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt (hunk 1 above)
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:14]: Fix

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt (hunk 3 above)
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt]: Fix

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt (hunk 4 above)
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt]: Thank you, fixed

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt (hunk 2 above)
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt]: Moved

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt [hunk 5]
```diff
@@ -10,44 +10,52 @@ import com.vk.admstorm.notifications.AdmNotification
...
+    private const val SSH_NOTIFICATOR_OPTION = "sshOpenSSHChanges"
     private const val SSH_CONFIG_BACKEND = "ssh.config.backend"
 
     private fun changeSshConfiguration(configValue: SshConnectionConfigService.Kind) {
         AdvancedSettings.setEnum(SSH_CONFIG_BACKEND, configValue)
     }
 
     fun changeConfigurationProcess(project: Project) {
-        val dntShow = PropertiesComponent.getInstance(project).getBoolean(SSH_NOTIFICATOR_OPTION)
+        val prpCmp = PropertiesComponent.getInstance(project)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:22]: What is `prpCmp`? Why not `propertiesComponent`, `properties`, `prop`? (I would use 'properties`)

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt [hunk 6]
```diff
@@ -10,44 +10,52 @@ import com.vk.admstorm.notifications.AdmNotification
...
-                    changeSshConfiguration(SshConnectionConfigService.Kind.OPENSSH)
-                    LOG.info("SSH option changed to OpenSSH")
+                    changeSshConfiguration(SshConnectionConfigService.Kind.LEGACY)
+                    LOG.info("SSH option changed to LEGACY")
                     notification.expire()
                 }
             ).show(project)
 
-        PropertiesComponent.getInstance(project).setValue(SSH_NOTIFICATOR_OPTION, true)
+        prpCmp.setValue(LEGACY_SSH_NOTIFICATOR_OPTION, true)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt:59]: Again, `LEGACY_SSH_NOTIFICATOR_OPTION`

File: src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt (hunk 2 above)
Comment (Danil42Russia) [reply src/main/kotlin/com/vk/admstorm/startup/ChangeSshBackendStartup.kt]: `isValueSet`, just checks if the key exists

**Review by Danil42Russia — APPROVED** [review]

**Pull request merged**